File: lecture6/cash.py
# get_float and print, with multiple arguments
from cs50 import get_float

def main():  # set function
    while True:  # while it is true 
        change = get_float("enter change: ")  # get input from user 
        if change > 0:  # break if input is greter than 0 
            break
    count = 0  # variable for numbers of coins 
    if change > 0:  # if change is greater than 0
        while change >= 0.25:  # while change is greater than 0.25 
            change -= 0.25  # subtract from 0.25
            count += 1  # add one to the count 
    
        while change >= 0.10:  # while change is greater than 0.10 
            change -= 0.10  # subtract from 0.10
            count += 1  # add one to the count 
    
        while change > 0.04:  # while change is greater than 0.05 
            change -= 0.05  # subtract from 0.05
            count += 1  # add one to the count 
    
        while change < 0.05 and change >= 0.01:  # while change is greater than 0.01 
            change -= 0.01  # subtract from 0.01
            count += 1  # add one to the count 
        print(count)  # print the numbers of coins 
    else:
        print(count)  #print count 

if __name__ == "__main__":  # state function 
    main()  # return to main 

File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int get_num(void);
int main(void)
{
    //assign variable for the number of coins 
    
    int value = get_num(); //get input for value 
    printf("%i\n", value);
// for loops to determine how many of each cent needed for the value input 
    int count = 0;
    for (int q = 1; value >= 25; q++) 
    {
        value -= 25; // keep subtracting until value is less than 0.25
        count++; //+1 to count everytime when subtracted 
    }
    for (int d = 1; value >= 10; d++)
    {
        value -= 10;
        count++;
    }
    for (int n = 1; value >= 5; n++)
    {
        value -= 5;
        count++;
    }
    for (int p = 1; value >= 1; p++)
    {
        value -= 1;
        count++;
    }
    printf("%i coins\n", count); //print count for user 
}

int get_num(void)
{
    float num;
    do
    {
        num = round((get_float("Change owed: \n")) * 100);
    }
    while (num < 0);
    return num;
}
